extensions/amp-base-carousel/1.0/base-element.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BaseCarousel} from './component';
+import {CSS as COMPONENT_CSS} from './component.jss';
+import {CarouselContextProp} from './carousel-props';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {dict} from '../../../src/core/types/object';
+
+export class BaseElement extends PreactBaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?number} */
+    this.slide_ = null;
+  }
+
+  /** @override */
+  init() {
+    const {element} = this;
+    this.slide_ = parseInt(element.getAttribute('slide'), 10);
+    return dict({
+      'defaultSlide': this.slide_ || 0,
+      'onSlideChange': (index) => {
+        this.triggerEvent(element, 'slideChange', dict({'index': index}));
+      },
+    });
+  }
+
+  /** @override */
+  mutationObserverCallback() {
+    const slide = parseInt(this.element.getAttribute('slide'), 10);
+    if (slide === this.slide_) {
+      return;
+    }
+    this.slide_ = slide;
+    if (!isNaN(slide)) {
```

Comment:
Probably not for this PR, but it looks like `.goToSlide()` could break if handed `NaN`? Perhaps it should handle that case (since I don't think type-checking can given `typeof NaN == 'number'`

Reply:
Thanks good point Ryan.  After chatting offline, I agree this is a good callout, could be a good candidate to surface an error message or have further functionality within `.goToSlide()`.

I'll defer to @caroqliu to comment if the `NaN` case is sufficiently handled or a possible use case requiring an error.

References within `.goToSlide`
https://github.com/ampproject/amphtml/blob/main/extensions/amp-base-carousel/1.0/base-carousel.js#L205
https://github.com/ampproject/amphtml/blob/main/extensions/amp-base-carousel/1.0/base-carousel.js#L184

Reply:
Can you clarify when `NaN` would break `goToSlide`? It does look handled in the above references but I could be missing something.

Reply:
I believe `NaN` is sufficiently handled so that it would not break `goToSlide`.

I was also wondering if it is a common enough use case to throw an error or if its sufficiently handled here with silent return.

Reply:
Merging for now, and will follow up offline to see if a follow up PR is needed!